app/controllers/api/v1/users_controller.rb

```diff
@@ -0,0 +1,15 @@
+class Api::V1::UsersController < ApplicationController
+  def show
+    if (username = params[:user][:username]) && (token = params[:user][:token])
+      user = User.find_or_create_by(uid: params[:id])
```

Comment:
pretty cool that it works like an enumerable even though it's missing some of the typical signs that it would be.